tests/rbd/rbd_snapshot_rename_advanced.py

```diff
@@ -0,0 +1,212 @@
+                imagespec=image_spec,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap,
+                new_snap_name=snap_rename,
+            )
+
+        if results["snap_rename"] or results["benchwrite"]:
+            log.error(f"Renaming snapshot while writing IOs failed for {snap_name}")
+            return 1
+
+        log.info(
+            "Rename the snapshot, when resize operation is happening on Clone image."
+        )
+        with parallel() as p:
+            p.spawn(
+                execute,
+                rbd,
+                "image_resize",
+                results,
+                pool_name=pool,
+                image_name=clone,
+                size=resize,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap_rename,
+                new_snap_name=f"{snap_rename}_1",
+            )
+
+        if results["snap_rename"] or results["image_resize"]:
+            log.error(
+                f"Renaming snapshot failed for {snap_name} while performing resize on clone {clone}"
+            )
+            return 1
+
+        log.info("Rename a snapshot, and then create a clone on top of it.")
+        clone_2 = f"{clone}_2"
+        snap_name = f"{pool}/{image}@{snap_rename}_1"
+        if rbd.create_clone(snap_name, pool, clone_2):
+            log.error(f"Creation of clone {clone_2} failed")
+            return 1
+
+        log.info(
+            "Rename the Parent snapshot, when Flatten operation is happening on Clone image."
+        )
+        with parallel() as p:
+            p.spawn(
+                execute,
+                rbd,
+                "flatten_clone",
+                results,
+                pool_name=pool,
+                image_name=clone_2,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=f"{snap_rename}_1",
+                new_snap_name=f"{snap_rename}_2",
+            )
+
+        if results["snap_rename"] or results["flatten_clone"]:
+            log.error(
+                f"Renaming snapshot failed for {snap_name} while performing flatten on clone {clone_2}"
+            )
+            return 1
+        return 0
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        if not kw.get("config").get("do_not_cleanup_pool"):
+            rbd.clean_up(pools=[pool])
+
+
+def run(**kw):
+    """Verification of snap and clone on an imported image.
```

Comment:
Are we really importing an image ?

Reply:
This was a line from a different test case, sorry for the confusion. I have changed the line to the relevant description now.